### PR TITLE
[SDL-0236] Update mismatch in TireStatus structure

### PR DESCRIPTION
**Pull request by VjKlepikov**
Implements [FORDTCN-12140](https://adc.luxoft.com/jira/browse/FORDTCN-12140)

This PR is **[ready]** for review. (internal review only)

### Summary
ATF scripts to cover SDL behavior with old applications for the 'Update mismatch in TireStatus structure' feature

### ATF version
[8.0.0_version_bump](https://github.com/smartdevicelink/sdl_atf/tree/feature/8.0.0_version_bump)

### Changelog
 - Add scripts to check default values for the old app

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: test_scripts/API/VehicleData/TireStatusMismatch/common.lua [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+m.getHMIConnection = actions.hmi.getConnection
+m.policyTableUpdate = actions.policyTableUpdate
+m.cloneTable = utils.cloneTable
+m.spairs = utils.spairs
+m.isTableEqual = utils.isTableEqual
+m.tableToString = utils.tableToString
+
+--[[ Common Variables ]]
+m.tirePressureParams = {
+  "pressureTelltale", "leftFront", "rightFront", "leftRear", "rightRear", "innerLeftRear", "innerRightRear"
```
Comment (atiwari9) [inline test_scripts/API/VehicleData/TireStatusMismatch/common.lua:34]: @VjKlepikov  - So are we sending all the params for default/non-default tests? Is there a test where we skip each of these params or more in signal OVD/GVD response?

Comment (atiwari9) [reply test_scripts/API/VehicleData/TireStatusMismatch/common.lua:34]: It seems the test 1 and 2 above only test non-mandatory params for `SingleTireStatus`. Can you please add the tests which omit params like `leftRear`, `leftFront` etc.

Comment (dboltovskyi) [reply test_scripts/API/VehicleData/TireStatusMismatch/common.lua:34]: @atiwari9 
Please notice delivery PR for review is `https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/2560`
Current PR was created just for Luxoft team internal review and was merged in main feature branch before delivery.

Regarding your comments:
Tests in current PR cover only cases for old applications (<=7.1)
In these scenarios omitted sub-parameters of `TireStatus` struct would be defaulted.

Existing VD tests cover all the cases for a new applications (>=8.0)
These tests would automatically take into account the changes in Mobile API ([rpc_spec](https://github.com/smartdevicelink/rpc_spec/compare/72632f946941d63a57ee5e99896e3eae3627f7dd...44f0a419c6a43815bc095134712eab4fef352ad7#diff-42ba46e1d01a876881e11902ad4a762c5040d0e7b089b17e7bfaaad9b10cd932) posted in `https://github.com/smartdevicelink/sdl_core/pull/3583`) and would generate appropriate scenarios.
In these scenarios omitted sub-parameters would be omitted as well in appropriate message for mobile application (response or notification)

As for new scripts in this PR:
- `001` and `002` - cover the following scenarios:
     - One sub-parameter is missing (iterating through all of them)
     - All sub-parameters are missing
     - All sub-parameters are present
 - `003` - covers scenario with 2 applications of different versions (`7.1` and `8.0`). In notification for old app omitted parameters would be defaulted, however for a new app those would be omitted

Comment (atiwari9) [reply test_scripts/API/VehicleData/TireStatusMismatch/common.lua:34]: Ok makes sense, i'd try to run the generic vehicle data test set to ensure that this is covered. Can you also please give that a run when possible.

**Comment by atiwari9** [comment]
[TestingReports_TirePressure.zip](https://github.com/LuxoftSDL/sdl_atf_test_scripts/files/6704174/TestingReports_TirePressure.zip)

Test Passed.

**Review by atiwari9 — COMMENTED** [review]
Approved

**Pull request merged**